bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandler.java

```diff
@@ -0,0 +1,344 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxtronikheatpump.internal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpChannel;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpCoolingOperationMode;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpOperationMode;
+import org.openhab.binding.luxtronikheatpump.internal.exceptions.InvalidChannelException;
+import org.openhab.binding.luxtronikheatpump.internal.exceptions.InvalidOperationModeException;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LuxtronikHeatpumpHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class LuxtronikHeatpumpHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LuxtronikHeatpumpHandler.class);
+    private final Set<ScheduledFuture<?>> scheduledFutures = new HashSet<>();
+    private static final int RETRY_INTERVAL = 60;
+    private boolean tiggerChannelUpdate = false;
+    private final LuxtronikTranslationProvider translationProvider;
+
+    public LuxtronikHeatpumpHandler(Thing thing, LuxtronikTranslationProvider translationProvider) {
+        super(thing);
+        this.translationProvider = translationProvider;
+    }
+
+    @Override
+    public void updateState(String channelID, State state) {
+        super.updateState(channelID, state);
+    }
+
+    @Override
+    public void updateProperty(String name, String value) {
+        super.updateProperty(name, value);
+    }
+
+    public void setStatusConnectionError() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
```

Comment:
As this is an expensive operation, you could store it once in `initialize()` in a field.

Reply:
moved it to the constructor. guess that should work as well.

Reply:
When the user changes the configuration during runtime, only `initialize()` is invoked, but not the constructor.

Reply:
makes sense. Would it be fine to load the config in constructor and overwrite it in initialize again? Otherwise I guess I would need to mark the field as nullable and check if it's set everytime it is used, or is there a smarter solution?

Reply:
You can simply initialize the field with an empty configuration to avoid the null checks.